### PR TITLE
Fix further test issues when run in container

### PR DESCRIPTION
**Pull request by m-horky**
* Card ID: ENT-5536

These tree commits address issues that were found when the test suite was run in a container.

UPDATE: The hwprobe commit was removed and will be opened individually, as it requires more testing.

### CODE REVIEW

File: test/rhsmlib/test_file_monitor.py [hunk 1]
```diff
@@ -35,6 +39,11 @@ def setUp(self):
         self.fsw1 = file_monitor.FilesystemWatcher(self.dir_list)
         self.fsw2 = file_monitor.InotifyFilesystemWatcher(self.dir_list)
 
+        now: float = time.time()
+        self.future = now + 2.0
+        for file in (self.testpath1, self.testpath2, self.testpath3):
+            os.utime(file, (now, now))
```
Comment (ptoscano) [inline test/rhsmlib/test_file_monitor.py:45]: hm do we need to change the times of the test files? they were created few lines before this, so they should be already "up to date"

File: test/rhsmlib/test_file_monitor.py [hunk 2]
```diff
@@ -35,6 +39,11 @@ def setUp(self):
         self.fsw1 = file_monitor.FilesystemWatcher(self.dir_list)
         self.fsw2 = file_monitor.InotifyFilesystemWatcher(self.dir_list)
 
+        now: float = time.time()
+        self.future = now + 2.0
```
Comment (ptoscano) [inline test/rhsmlib/test_file_monitor.py:43]: hm maybe 2 seconds is not much, in case the tests take more than that to run -- maybe bump it to 60 or even 120?

File: src/rhsmlib/facts/hwprobe.py [hunk 3]
```diff
@@ -782,20 +783,28 @@ def get_network_info(self) -> Dict[str, str]:
...
-                net_info["network.ipv4_address"] = ", ".join(ip_list)
-            except Exception as err:
-                log.debug("Error during resolving IPv4 address of hostname: %s, %s" % (hostname, err))
+            ipv4_addresses: Set[str] = set()
+            ipv6_addresses: Set[str] = set()
+            for address_tuple in infolist:
+                address: str = address_tuple[4][0]
+                if address_tuple[0] is socket.AF_INET:
+                    ipv4_addresses.add(address)
+                if address_tuple[0] is socket.AF_INET6:
```
Comment (ptoscano) [inline src/rhsmlib/facts/hwprobe.py:792]: minor nit: you can use `elif` here, as this condition cannot be true when the former is (and vice versa)

File: src/rhsmlib/facts/hwprobe.py [hunk 4]
```diff
@@ -782,20 +783,28 @@ def get_network_info(self) -> Dict[str, str]:
...
+                    ipv6_addresses.add(address)
+
+            # FIXME In 2017-09 (616ec72), we added `_get_ipvX_addr_list()` functions, but the original code
+            #  using sockets stayed; the new one was used just in the `except:` block. Why? Can't we just use
+            #  ethtool for everything? It is hard dependency and should always be available.
```
Comment (ptoscano) [inline src/rhsmlib/facts/hwprobe.py:797]: sadly [python-ethtool](https://github.com/fedora-python/python-ethtool) seems to be deprecated too:
> This is the new upstream for python-ethtool maintained by Fedora's Python SIG. We ported it to Python 3 and only maintain it for the current tools to keep working. No new development is happening. This is a deprecated package. If you are considering to start using this, please don't. We recommend [netifaces](https://pypi.org/project/netifaces/) instead. Another alternative is [ethtool parsers](https://insights-core.readthedocs.io/en/latest/shared_parsers_catalog/ethtool.html) from [insights-core](https://pypi.org/project/insights-core/).

maybe it'd be better to use `ip` (via its JSON output) for the network-related facts, assuming it provides all the details we currently provide as facts

**Review by ptoscano — CHANGES_REQUESTED** [review]
In general it is OK, thanks for the work!

While I agree with the change wrt the IP addresses collection, I think that that bit need a bit more of discussion and testing, especially that tried to handle some kind of "edge cases". Would it be possible to split that commit in its own draft PR, so we can bring the other two commits in?

**Review by ptoscano — APPROVED** [review]
Thanks!
